source/h1_connection.c

```diff
@@ -1066,6 +1080,27 @@ static int s_decoder_on_header(const struct aws_h1_decoded_header *header, void
                 connection->synced_data.new_stream_error_code = AWS_ERROR_HTTP_CONNECTION_CLOSED;
                 aws_h1_connection_unlock_synced_data(connection);
             } /* END CRITICAL SECTION */
+
+            /* For server side, server should finish sending the response. */
+            if (connection->base.client_data) {
```

Comment:
trivial: this comment confused me, more than helped me

I expect a comment before an if-statement to summarize what the if-statement is about
But this comment is explaining the else-case, before I've read what the if-statement even says.